homeassistant/components/wmspro/generic_entity.py

```diff
@@ -0,0 +1,46 @@
+"""Generic entity for the WMS WebControl pro API integration."""
+
+from __future__ import annotations
+
+from wmspro.destination import Destination
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+
+class WebControlProGenericEntity(Entity):
+    """Foundation of all WMS based entities."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, config_entry_id: str, dest: Destination) -> None:
```

Comment:
Personal preference, switch them around since `dest` is more important than the configentryid

Reply:
My personal preference here is: HA related arguments before custom arguments.

Reply:
And also: parameter scope decreases from start to finish, e.g. hub first, device second.